engine/src/main/java/org/terasology/logic/debug/MovementDebugCommands.java

```diff
@@ -203,4 +217,23 @@ public String stepHeight(@Sender EntityRef client, @CommandParam("height") float
 
         return "";
     }
+
+    @Command(shortDescription = "Sets the height of the player", runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String playerHeight(@Sender EntityRef client, @CommandParam("height") float amount) {
+        ClientComponent clientComp = client.getComponent(ClientComponent.class);
+        CharacterMovementComponent move = clientComp.character.getComponent(CharacterMovementComponent.class);
+        if (move != null) {
+            float prevHeight = move.height;
+            move.height = amount;
+            clientComp.character.saveComponent(move);
+            LocationComponent loc = client.getComponent(LocationComponent.class);
+            Vector3f currentPosition = loc.getWorldPosition();
+            teleportCommand(client, currentPosition.getX(), currentPosition.getY() + (amount - prevHeight)/2, currentPosition.getZ());
```

Comment:
instead of using the other command go for `CharacterTeleportEvent`(as in the teleport command itself). Two reasons for this: 
(a) the visible effect of sorts should be a change of player height, I don't bother whether the character needed to be teleported, and
(b) it should be easier to refactor this later on when you want to integrate it with potion effects and such.